### PR TITLE
feat: Create Breadcrumbs component

### LINKED ISSUE #1736: DOJO - Create Breadrcumbs component
[Component on Figma](https://www.figma.com/file/f9xRyl1bJOjtBRd8zFORzg/c0d3?node-id=311%3A2147)

image

Could use [react-bootstrap breadcrumbs](https://react-bootstrap.github.io/components/breadcrumb/#breadcrumbs)

### PR DESCRIPTION
**Pull request by flacial**
Closes #1736 
Depends on #1753

This PR creates the Breadcrumbs component using the following [design](https://www.figma.com/file/f9xRyl1bJOjtBRd8zFORzg/c0d3?node-id=204%3A4535) that will be used to make navigation easier on Admin and Mentor pages.

Demo:

image

### CODE REVIEW

File: components/Breadcrumbs.test.js [hunk 1]
```diff
@@ -0,0 +1,105 @@
+import React from 'react'
+import { render } from '@testing-library/react'
+import Breadcrumbs from './Breadcrumbs.tsx'
+import '@testing-library/jest-dom'
```
Comment (flacial) [inline components/Breadcrumbs.test.js:4]: Imported to be able to use `expect(...).toBeInTheDocument() `

File: components/Breadcrumbs.test.js [hunk 2]
```diff
@@ -0,0 +1,105 @@
...
+import userEvent from '@testing-library/user-event'
+
+const useRouter = jest.spyOn(require('next/router'), 'useRouter')
```
Comment (flacial) [inline components/Breadcrumbs.test.js:7]: I found a working way to mock `useRouter` @SlyBouhafs

File: components/Breadcrumbs.tsx [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+        </span>
+      )}
+    </li>
+  )
+
+  const homeBreadcrumb = !omitHomeRoute && (
+    <Breadcrumb index={0} breadcrumb={{ title: homeTitle, href: '/' }} />
+  )
+
+  // TODO: Specific code for Admin page that needs to be refactored
```
Comment (flacial) [inline components/Breadcrumbs.tsx:84]: In the last engineering meeting, we voted on not making it generic. It'll be in a future PR.

File: components/Breadcrumbs.test.js (hunk 1 above)
Comment (hwong0305) [reply components/Breadcrumbs.test.js:4]: Add this as a comment in the file.

File: components/Breadcrumbs.tsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+
+type Props = {
+  omitHomeRoute?: boolean
+  homeTitle?: string
+  lessonTitle: string
+  setLessonTitle: Dispatch<SetStateAction<string>>
+  lessons: GetAppQuery['lessons'] | undefined
+}
+
+type Breadcrumb = {
```
Comment (hwong0305) [inline components/Breadcrumbs.tsx:18]: Is the rest of codebase using `type` or `interface`. Whichever ever it is, let's make sure we pick one and be consistent.

Comment (anthonykhoa) [reply components/Breadcrumbs.tsx:18]: Are you suggesting to stick to using just `type` or `interface` across the whole codebase in every scenario? Or are you suggesting to stick to using `type` or `interface` only when declaring type definitions for objects?

**Review by songz — APPROVED** [review]

**Review by SlyBouhafs — APPROVED** [review]

**Pull request merged**